src/proj_4D_api.c

```diff
@@ -1004,7 +1008,7 @@ PJ_INIT_INFO proj_init_info(const char *initname){
     key[64] = 0;
     strncat(key, ":metadata", 9);
     strcpy(param, "+init=");
-    strncat(param, key, 73);
+    strncat(param + strlen(param), key, sizeof(param)-1-strlen(param));
```

Comment:
I don't understand the `param + strlen(param)` part of this. What's going on here?

Reply:
This helps cppcheck (probably false positive otherwise). strncat(foo, bar, baz) is equivalent of strncat(foo + strlen(foo), baz, baz)
The real fix is to make sure to use sizeof(foo) - 1 - strlen(foo) to make sure it doesn't go beyond end of buffer

Reply:
Ah, okay. I think a comment here is warranted. At least I'll forget this explanation quickly and wonder what's going on here the next time I stop by.

Reply:
Comment added

Reply:
Thanks